runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -209,7 +208,7 @@ public void gc(long trimMark) {
                 if (directAccessCheckFunction.apply(this)) {
                     log.trace("Access [{}] Direct (optimistic-read) access at {}",
                             this, getVersionUnsafe());
-                    R ret = accessFunction.apply(object);
+                    R ret = accessFunction.apply(object.getContext(ICorfuExecutionContext.DEFAULT));
```

Comment:
Might be missing something, but what's the point of staging/upstaging uncommitted changes, if the access methods can never access the upstaged context? Essentially all that work is unnecessary, since it has no consequence on the read path. 

It seems like this PR introduces changes to implement some sort of read-committed semantics, but in many systems read committed actually returns upstaged changes (created by yourself/same transaction), so this seems a bit strange: you can't read you're own writes?  

You can easily support read-your-write semantics by re-routing the access function to either the `ICorfuExecutionContext.DEFAULT`, or `ICorfuExecutionContext.Optimistic`

Reply:
You are not missing anything. The plan is to implement read-your-own-dirty-writes, but since the current clients do not care about this type of behavior, I left it as unimplemented.

This will have to be implemented as a two-phase lookup.